pilz_control/test/unittest_pilz_joint_trajectory_controller.cpp

```diff
@@ -382,159 +260,334 @@ TEST_F(PilzJointTrajectoryControllerTest, testDoubleRequest)
+ */
+class PilzJointTrajectoryControllerIsExecutingTest : public PilzJointTrajectoryControllerTest,
+                                                     public testing::WithParamInterface<InvokeIsExecuting>
+{
+protected:
+  testing::AssertionResult invokeIsExecuting();
+  testing::AssertionResult waitForIsExecutingResult(bool expectation);
+
+  /**
+   * @brief Perform init, start, unhold and update, such that controller is ready for executing.
+   */
+  testing::AssertionResult performFullControllerStartup();
+
+  bool is_executing_result;
+};
+
+testing::AssertionResult PilzJointTrajectoryControllerIsExecutingTest::invokeIsExecuting()
+{
+  auto invoke_is_executing {GetParam()};
+  return invoke_is_executing(controller_, is_executing_result);
+}
+
+testing::AssertionResult PilzJointTrajectoryControllerIsExecutingTest::waitForIsExecutingResult(bool expectation)
+{
+  auto invoke_is_executing {GetParam()};
+  ros::Duration update_period{DEFAULT_UPDATE_PERIOD_SEC};
+
+  while (ros::ok())
+  {
+    progressInTime(update_period);
+    updateController();
+
+    auto assertion_result {invoke_is_executing(controller_, is_executing_result)};
+    if(assertion_result == testing::AssertionFailure())
+    {
+      return assertion_result;
+    }
+    if (is_executing_result == expectation)
+    {
+      return testing::AssertionSuccess();
+    }
+
+    usleep(SLEEP_TIME_MSEC);
+  }
+
+  return testing::AssertionFailure()
+      << "Controller did not " << (expectation ? "start" : "stop") << " executing as expected.";
+}
+
+testing::AssertionResult PilzJointTrajectoryControllerIsExecutingTest::performFullControllerStartup()
+{
+  if (!controller_->init(hardware_, nh_, controller_nh_))
+  {
+    return testing::AssertionFailure() << "Failed to initialize the controller.";
+  }
+  controller_->state_ = controller_->INITIALIZED;
+
+  startController();
+  controller_->state_ = controller_->RUNNING;
+
+  std_srvs::TriggerRequest req;
+  std_srvs::TriggerResponse resp;
+  if (!(controller_->handleUnHoldRequest(req, resp) && resp.success))
+  {
+    return testing::AssertionFailure() << "Unholding the controller failed.";
+  }
+
+  ros::Duration stop_duration{STOP_TRAJECTORY_DURATION + DEFAULT_UPDATE_PERIOD_SEC};
+  progressInTime(stop_duration);
+  updateController();
+
+  return testing::AssertionSuccess();
+}
+
+TEST_P(PilzJointTrajectoryControllerIsExecutingTest, testNotInitialized)
+{
+  EXPECT_TRUE(invokeIsExecuting());
+  EXPECT_FALSE(is_executing_result) << "Controller is executing unexpectedly";
+}
+
+TEST_P(PilzJointTrajectoryControllerIsExecutingTest, testNotStarted)
+{
+  ASSERT_TRUE(controller_->init(hardware_, nh_, controller_nh_)) << "Failed to initialize the controller.";
+  controller_->state_ = controller_->INITIALIZED;
+
+  EXPECT_TRUE(invokeIsExecuting());
+  EXPECT_FALSE(is_executing_result) << "Controller is executing unexpectedly";
+}
+
+/**
+ * @brief Test the is_executing functionality of the started controller.
```

Comment:
All test documentations are written as if we want to test the `is_executing` function. But in my opinion, we actually want to test the different controller functionalities whereas the `is_executing` function is just a mean to achieve this. Therefore, I suggest to describe what controller functionality the test is supposed to test instead.
What do you think?

**Please note:**
The tests are now much shorter. So, in my opinion, it is enough to provide a good short description of what the test wants to test and then skip/delete the detailed test-step descriptions.

**Please note:**
This comment holds for all test documentations in this test suite.

Reply:
Interesting. I had the opinion that we test the `is_executing` functionality. But it is a matter of point of view.

Reply:
I introduced short descriptions. Done.

Reply:
Will merge as soon as build checks are through.